### ISSUE TITLE
[Bug]: On Android 12 and lower, it never returns PermissionStatus.permanentlyDenied for notifications

### ISSUE THREAD

**Issue by tom-vanlieshout** [Baseflow/flutter-permission-handler#1244]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-permission-handler/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-permission-handler/blob/main/permission_handler/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [ ] iOS
- [ ] Windows

### Steps to reproduce

1. Use the Flutter example app and edit the onTap handler to: final response = await Permission.notification.request();
2. Optionally, add a print-statement to log the result of `response`
3. Go into the system settings of the device, navigate to Apps, open settings for the example app and turn off permission for notifications
4. Go back into the example app and hit the onTap-handler again, multiple times
5. Observe the app always logging `PermissionStatus.denied`

### Expected results

The first time I'd expect `PermissionStatus.denied`, every other time after that I'd expect `PermissionStatus.permanentlyDenied`.

### Actual results

The PermissionStatus is always `PermissionStatus.denied`.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:permission_handler/permission_handler.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});
  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  void _incrementCounter() async {
    print(await Permission.notification.status);
    final response = await Permission.notification.request();
    print(response);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: const Text("Body"),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}

```

</details>

### Version

11.0.1

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
tomvanlieshout@MacBook-Pro sandbox % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.0, on macOS 14.0 23A344 darwin-arm64, locale en-NL)
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.84.2)
[✓] VS Code (version 1.80.2)
[✓] Connected device (3 available)
[✓] Network resources

! Doctor found issues in 1 category.
```

</details>

<details><summary>pubspec.yaml</summary>

```yaml
name: sandbox
description: A new Flutter project.

publish_to: 'none' # Remove this line if you wish to publish to pub.dev

version: 1.0.0+1

environment:
  sdk: '>=2.18.6 <3.0.0'

dependencies:
  cupertino_icons: ^1.0.2
  flutter:
    sdk: flutter
  permission_handler: ^11.0.1

dev_dependencies:
  flutter_lints: ^2.0.0
  flutter_test:
    sdk: flutter

flutter:
  uses-material-design: true

```

</details>

### **Additional info**

I've tested this on a OnePlus 6 with Android 11, on a Samsung device with Android 10. Both show this (probably) incorrect behaviour. It does return permanentlyDenied on Android 14, Android 13 and on iOS.

The issue is that I'd like to request my user through a popup which only works the very first time we request permission. Any time after that, the OS does not allow the popup and the only thing we can do, is bring the user to the system settings. I'm determining whether to ask through popup or system settings based on the `PermissionStatus`, hence it is important that this works the same across different OS and versions.

Furthermore, I'd expect that if the status is `PermissionStatus.denied` I can request permission through the popup. This also does not work on Android 12 or lower. Only opening the OS settings works. I'm opening this issue because I think code implementation should not depend so much on OS versions.

**Comment by TimHoogstrate** [comment]
Dear @tom-vanlieshout,

If I compare change logs it seems that this has been fixed in version: 11.1.0. I'll close this issue for now, feel free to reopen the issue when new information becomes available. 

Kind regards,

**Comment by joshminton** [comment]
I'm still getting this behaviour with 11.3.1.

**Comment by jamieastley** [comment]
@TimHoogstrate this shouldn't have been closed, this is still reproducible on 11.3.1.
